Update pip to pip3 in Dockerfiles

michhar opened the pull request:
Dockerfiles were using pip instead of pip3 and this was causing failure for Python (now Python 3) modules to run properly - as the proper packages were not installed into the image.  Therefore, instances of `pip` were updated to `pip3` and the `Dockerfile.amd64` produces a successfully running module using the VSCode pathway to building and deploying an IoT Edge module.

michhar commented:
Please advise on why this build failed.  I do not have access to the "Details".  I'm happy to correct it.

formulahendry commented:
Hi @michhar , I could successfully build and run the Python module. Could you share the detailed error message and repro steps?
![image](https://user-images.githubusercontent.com/1050213/47631889-daed0d00-db82-11e8-861d-cdee64175cb4.png)

michhar commented:
Hi, thanks for your response.  I've repeated your results and all is now working for me (although I updated the base images to 1.0.4).  I even added a library to test if pip was installing into the python3 environment.  Both work.  This issue may be closed.

The repro:

1.  Perform steps in this tutorial:  https://docs.microsoft.com/en-us/azure/iot-edge/tutorial-python-module
2.  Add "import flask" to top of `main.py`.
3.  Add "flask==1.0.2" to `requirements.txt`

Updated `deployment.template.json`:
```json
{
  "modulesContent": {
    "$edgeAgent": {
      "properties.desired": {
        "schemaVersion": "1.0",
        "runtime": {
          "type": "docker",
          "settings": {
            "minDockerVersion": "v1.25",
            "loggingOptions": "",
            "registryCredentials": {
              "michharregistry": {
                "username": "$CONTAINER_REGISTRY_USERNAME_michharregistry",
                "password": "$CONTAINER_REGISTRY_PASSWORD_michharregistry",
                "address": "michharregistry.azurecr.io"
              }
            }
          }
        },
        "systemModules": {
          "edgeAgent": {
            "type": "docker",
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-agent:1.0.4",
              "createOptions": "{}"
            }
          },
          "edgeHub": {
            "type": "docker",
            "status": "running",
            "restartPolicy": "always",
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-hub:1.0",
              "createOptions": "{\"HostConfig\":{\"PortBindings\":{\"5671/tcp\":[{\"HostPort\":\"5671\"}], \"8883/tcp\":[{\"HostPort\":\"8883\"}],\"443/tcp\":[{\"HostPort\":\"443\"}]}}}"
            }
          }
        },
        "modules": {
          "tempSensor": {
            "version": "1.0",
            "type": "docker",
            "status": "running",
            "restartPolicy": "always",
            "settings": {
              "image": "mcr.microsoft.com/azureiotedge-simulated-temperature-sensor:1.0.4",
              "createOptions": "{}"
            }
          },
          "SampleModule": {
            "version": "1.0",
            "type": "docker",
            "status": "running",
            "restartPolicy": "always",
            "settings": {
              "image": "${MODULES.SampleModule.amd64}",
              "createOptions": "{}"
            }
          }
        }
      }
    },
    "$edgeHub": {
      "properties.desired": {
        "schemaVersion": "1.0",
        "routes": {
          "SampleModuleToIoTHub": "FROM /messages/modules/SampleModule/outputs/* INTO $upstream",
          "sensorToSampleModule": "FROM /messages/modules/tempSensor/outputs/temperatureOutput INTO BrokeredEndpoint(\"/modules/SampleModule/inputs/input1\")"
        },
        "storeAndForwardConfiguration": {
          "timeToLiveSecs": 7200
        }
      }
    },
    "SampleModule": {
      "properties.desired":{
          "TemperatureThreshold":25
      }
  }
  }
}
```
